chef-utils/lib/chef-utils/dsl/path_sanity.rb

```diff
@@ -30,27 +30,27 @@ def sanitized_path(env = nil)
         path_separator = ChefUtils.windows? ? ";" : ":"
         # ensure the Ruby and Gem bindirs are included for omnibus chef installs
         new_paths = env_path.split(path_separator)
-        [ ChefUtils::DSL::PathSanity.ruby_bindir, ChefUtils::DSL::PathSanity.gem_bindir ].compact.each do |path|
+        [ __ruby_bindir, __gem_bindir ].compact.each do |path|
           new_paths = [ path ] + new_paths unless new_paths.include?(path)
         end
-        ChefUtils::DSL::PathSanity.sane_paths.each do |path|
+        __sane_paths.each do |path|
           new_paths << path unless new_paths.include?(path)
         end
         new_paths.join(path_separator).encode("utf-8", invalid: :replace, undef: :replace)
       end
 
-      class << self
-        def sane_paths
-          ChefUtils.windows? ? %w{} : %w{/usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin}
-        end
+      private
 
-        def ruby_bindir
-          RbConfig::CONFIG["bindir"]
-        end
+      def __sane_paths
```

Comment:
I know this is not all new code, but it'd be nice if we could avoid new uses of sane/sanity.

Reply:
Also is this method called outside of this module? If not, it could be a constant.

Reply:
its a "dependency injection hook" so it can't be a constant and is sort of a public API.  instead of constructor injection we override the inherited methods in the "subclass" where its wired up to.

i don't know how much people have discovered "path sanity" and started to use it externally as a public API thing.  we can change it, but i think i'd prefer to do that deliberately as a separate issue.  we should probably batch that up with the allow/deny_list stuff for 17.0 or something.

Reply:
(the "path_sanity" term dates back to like chef 0.8.x days or something along those lines)

Reply:
We're going to need to maintain those existing names but mark them deprecated and just ship new methods. Sooner we drop those names the better